### PR TITLE
Add entity Count functions

### LINKED ISSUE #125: Support entityCount functions
File
- [x] blockCount()
- [x] sectionCount()

Block
- [x] sourceCount()
- [x] dataArrayCount()
- [x] tagCount()
- [x] multiTagCount()
- [x] groupCount()

Group
- [x] dataArrayCount()
- [x] tagCount()
- [x] multiTagCount()
- [x] sourceCount()

DataArray
- [x] dimensionCount()
- [x] sourceCount()

Source
- [x] sourceCount()

Tag
- [x] referenceCount()
- [x] featureCount()
- [x] sourceCount()

MultiTag
- [x] referenceCount()
- [x] featureCount()
- [x] sourceCount()

Section
- [x] sectionCount()
- [x] propertyCount()

Property
- [x] valueCount()

### PR DESCRIPTION
**Pull request by mpsonntag**
Closes #125.

Successfully built under win32 (Matlab R2011a) and win64 (Matlab R2011a).

### CODE REVIEW

**Comment by jgrewe** [comment]
Noticed that we have two ways of naming things in the API. For example in https://github.com/mpsonntag/nix-mx/blob/caa637a1584c1054796f35b692186adb81e2dd87/%2Bnix/File.m many methods use the camelCase while most others use the under_score. 
We should unify this. (just found another one https://github.com/mpsonntag/nix-mx/blob/caa637a1584c1054796f35b692186adb81e2dd87/%2Bnix/Section.m#L60)

**Comment by gicmo** [comment]
@jgrewe I think Andrey, Michi and I had even a discussion about the naming thing. It is a bit unfortunate because MATLAB uses snake case but in we in NIX cpp use camelCase. I think the consensus was to use snake_case but I am not sure.

**Comment by mpsonntag** [comment]
The naming issue on the matlab side has been on my mind as well since they are mixed since time immemorial. For now I try to stick to the naming conventions as they are in each matlab file but would like to do a one time pull request renaming every function name on the matlab side to snake case.

File: nix_mx.cc [hunk 1]
```diff
@@ -113,7 +113,9 @@ void mexFunction(int            nlhs,
             .reg("openBlock", GETBYSTR(nix::Block, nix::File, getBlock))
             .reg("openSection", GETBYSTR(nix::Section, nix::File, getSection))
             .reg("createBlock", &nix::File::createBlock)
-            .reg("createSection", &nix::File::createSection);
+            .reg("createSection", &nix::File::createSection)
+            .reg("blockCount", GETTER(unsigned long long int, nix::File, blockCount))
```
Comment (gicmo) [inline nix_mx.cc]: Can't we use `nix::ndsize_t` here (and all below)?

Comment (mpsonntag) [inline nix_mx.cc]: Aaah, was not aware of `nix::ndsize_t`. I shall use it, since this one does not crash the build! ;)

**Comment by gicmo** [comment]
LGTM

**Pull request merged**